hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/DirectoryEntityHandler.java

```diff
@@ -89,7 +91,13 @@ public DUResponse getDuResponse(
     for (long subdirObjectId: subdirs) {
       NSSummary subdirNSSummary =
               getReconNamespaceSummaryManager().getNSSummary(subdirObjectId);
-      String subdirName = subdirNSSummary.getDirName();
+      // for the subdirName we need the filename, not the key name
+      Path subdirPath = Paths.get(subdirNSSummary.getDirName());
+      Path subdirFileName = subdirPath.getFileName();
+      String subdirName = "";
+      if (subdirFileName != null) {
+        subdirName += subdirFileName.toString();
+      }
```

Comment:
not sure what is going on here, but it doesn't look right.  if the subdirfilename is null, then this is going to be wrong, isn't it?